codec-http2/src/main/java/io/netty/handler/codec/http2/Http2StreamFrame.java

```diff
@@ -40,5 +44,12 @@
      * applies to a particular stream, or a value less than {@code 0} if the frame has yet to be associated with
      * the connection or a stream.
      */
-    int streamId();
+    int getStreamId();
+
+    /**
+     * Returns {@code true} if the frame has a valid stream identifier set, {@code false} otherwise.
+     *
+     * <p>Once this method has returned {@code true}, it must never return {@code false} again.
+     */
+    boolean hasStreamId();
```

Comment:
why we even need this ? Couldn`t we just specify that `-1` signals no stream id ?